control-plane-pki-operator/certrotationcontroller/certrotationcontroller.go

```diff
@@ -0,0 +1,153 @@
+package certrotationcontroller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	hypershiftv1beta1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	hypershiftv1beta1client "github.com/openshift/hypershift/client/clientset/clientset/typed/hypershift/v1beta1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/control-plane-pki-operator/clienthelpers"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/library-go/pkg/controller/factory"
+
+	"github.com/openshift/library-go/pkg/operator/certrotation"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// defaultRotationDay is the default rotation base for all cert rotation operations.
+const defaultRotationDay = 24 * time.Hour
+
+type CertRotationController struct {
+	certRotators []factory.Controller
+
+	recorder events.Recorder
+
+	cachesToSync []cache.InformerSynced
+}
+
+func NewCertRotationController(
+	hostedControlPlane *hypershiftv1beta1.HostedControlPlane,
+	kubeClient kubernetes.Interface,
+	hypershiftClient hypershiftv1beta1client.HostedControlPlanesGetter,
+	configInformer configinformers.SharedInformerFactory,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	eventRecorder events.Recorder,
+	day time.Duration,
+) (*CertRotationController, error) {
+	ret := &CertRotationController{
+		recorder: eventRecorder,
+		cachesToSync: []cache.InformerSynced{
+			configInformer.Config().V1().Networks().Informer().HasSynced,
+			configInformer.Config().V1().Infrastructures().Informer().HasSynced,
+		},
+	}
+
+	rotationDay := defaultRotationDay // TODO(deads2k): is this logic we want to persist? looks temporary ...
```

Comment:
@deads2k

Reply:
something similar is important to have here for testing with tighter timeframes.  I suggest keeping it.

Reply:
Moreso the question was - it looks like if you don't specify something custom in options, you get the dev duration. If you *do* specify something, though, you get a log warning about an unsupported value - that seems incorrect?